lib/internal/modules/cjs/loader.js

```diff
@@ -654,15 +617,23 @@ Module._findPath = function(request, paths, isMain) {
   let trailingSlash = request.length > 0 &&
     request.charCodeAt(request.length - 1) === CHAR_FORWARD_SLASH;
   if (!trailingSlash) {
-    trailingSlash = /(?:^|\/)\.?\.$/.test(request);
+    trailingSlash = trailingSlashRegex.test(request);
   }
 
   // For each path
   for (let i = 0; i < paths.length; i++) {
     // Don't search further if path doesn't exist
     const curPath = paths[i];
     if (curPath && stat(curPath) < 1) continue;
-    const basePath = resolveExports(curPath, request, absoluteRequest);
+
+    if (!absoluteRequest) {
+      const exportsResolved = resolveExports(curPath, request);
+      if (exportsResolved) {
+        return exportsResolved;
+      }
+    }
+
+    const basePath = path.resolve(curPath, request);
```

Comment:
will `delete require('path').resolve` break this code?

Reply:
Yes, although there are a lot of calls to `path.resolve` within this module, so changing that is best done as a separate refactoring.